### ISSUE TITLE
igx-carousel accessibility improvements

### ISSUE THREAD

**Issue by kdinev** [IgniteUI/igniteui-angular#8202]
## Is your feature request related to a problem? Please describe.
The current `aria` roles applied on the `igx-carousel` widget fail in an accessibility scan. The carousel is marked as `role="list"`, but slides are not marked as list items. 

## Describe the solution you'd like
The carousel should be handled as a `tab` widget, where the buttons to choose a slide are `tabs` and the slides are `tabpanels`.

## Additional context
Put a carousel component in a document and run the lighthouse scan to check the a11y failures.

**Comment by zdrawku** [comment]
[Specification update with latest a11y changes ](https://github.com/IgniteUI/igniteui-angular/wiki/Carousel-Specification/_compare/71dbd3ea69581fc36ababb5c288325db61d30cbf...315426ed7b4c616668646ed08d53ecfb0f01c141)
